src/google/protobuf/compiler/csharp/csharp_message_field.cc

```diff
@@ -104,7 +104,7 @@ void MessageFieldGenerator::GenerateMergingCode(io::Printer* printer) {
     variables_,
     "if (other.$has_property_check$) {\n"
     "  if ($has_not_property_check$) {\n"
-    "    $property_name$ = new $type_name$();\n"
+    "    $property_name$ = $type_name$.Parser.CreateTemplate();\n"
```

Comment:
after changing the codegen, you'd also need to regenerate the code in this repository.

Reply:
= doing a build ?

Reply:
Build and then run https://github.com/protocolbuffers/protobuf/blob/master/csharp/generate_protos.sh.

The updates to the generated files need to be commited (preferrably as a separate commit)